thrift/thrift_test.go

```diff
@@ -250,8 +257,9 @@ func TestRegisterPostResponseCB(t *testing.T) {
 		}
 
 		called := make(chan struct{})
-		cb := func(method string, response thrift.TStruct) {
+		cb := func(reqCtx context.Context, method string, response thrift.TStruct) {
 			assert.Equal(t, "Call", method)
+			assert.Equal(t, ctx.Value(key), value)
```

Comment:
The issue is that `reqCtx` is not being tested -- you could pass `nil` as a parameter to the context in the Thrift code, and the test would still pass.

The test should make sure that `reqCtx` is the same context that is returned from the context creation function set by `SetContextFn`